File: admin-api/src/mujahid_admin/services/audit.py
from __future__ import annotations

from mujahid_admin.repositories import audit as audit_repo
from mujahid_admin.schemas.audit import AuditEntry, PaginatedAudit


def _paginate(page: int, page_size: int, total: int) -> int:
    return max((total + page_size - 1) // page_size, 1)


async def list_audit(page: int, page_size: int) -> PaginatedAudit:
    items, total = await audit_repo.list_audit(page, page_size)
    return PaginatedAudit(
        items=[
            AuditEntry(
                action=item["action"],
                target=item["target"],
                at=item["at"].isoformat() if item["at"] else None,
                ip=item["ip"],
            )
            for item in items
        ],
        page=page,
        page_size=page_size,
        total=total,
        total_pages=_paginate(page, page_size, total),
    )

File: admin-api/src/services/audit.ts
import * as auditRepo from "@/repositories/audit";
import type { PaginatedAudit } from "@/types";

function paginate(page: number, pageSize: number, total: number): number {
  return Math.max(Math.ceil(total / pageSize), 1);
}

export async function listAudit(page: number, pageSize: number): Promise<PaginatedAudit> {
  const [items, total] = await auditRepo.listAudit(page, pageSize);
  return {
    items: items.map((item) => ({
      action: item.action,
      target: item.target,
      at: item.at ? item.at.toISOString() : null,
      ip: item.ip,
    })),
    page,
    pageSize,
    total,
    totalPages: paginate(page, pageSize, total),
  };
}
